### ISSUE TITLE
Add requirements.txt with minimum requirements necessary.

### ISSUE THREAD

**Issue by rettigl** [OpenCOMPES/sed#49]
I cannot install the package with poetry:

```
Installing dependencies from lock file
Warning: The lock file is not up to date with the latest changes in pyproject.toml. You may be getting outdated dependencies. Run update to update them.

  SolverProblemError

  Because sed depends on ipykernel (^6.9.1) which doesn't match any versions, version solving failed.

  at /mnt/pcshare/users/Laurenz/AreaB/sed/.pyenv/lib/python3.8/site-packages/poetry/puzzle/solver.py:241 in _solve
      237│             packages = result.packages
      238│         except OverrideNeeded as e:
      239│             return self.solve_in_compatibility_mode(e.overrides, use_latest=use_latest)
      240│         except SolveFailure as e:
    → 241│             raise SolverProblemError(e)
      242│
      243│         results = dict(
      244│             depth_first_search(
      245│                 PackageNode(self._package, packages), aggregate_package_nodes

```
Also doing a pip -E install results in some dependency problems, it appears because dask is pinned to a specific version:
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
distributed 2022.2.1 requires dask==2022.02.1, but you have dask 2021.12.0 which is incompatible.
```

 Why is this the case? I would remove any specific version pinning, if not absolutely required.

**Comment by rettigl** [comment]
Okay, I understand now. 
`dask=^2021.12.0`
does not allow to jumpt to any 2022 version. In this case where they advance by year, we have to use >=.

**Comment by zainsohail04** [comment]
Could you try once more, the lock file didn't merge properly last time.

**Comment by daviddoji** [comment]
Anyway, I think it also makes sense to create a `requirements.txt` with pinned version.

**Comment by steinnymir** [comment]
did anyone test this before closing it? I am having a really hard time to get the requirements installed together with the requirements of an other package (in this case MPES).

The requirements.txt is forcing conflicting versions, and therefore fails.
The poetry installation is still very counterintuitive to me, I cannot seem to find a way to make it work as intended.

I will continue trying, but this issue should be closed only once a well tested environment installation procedure has been identified and described.

Also, a conda install .yaml would be really helpful.

**Comment by zainsohail04** [comment]
> did anyone test this before closing it? I am having a really hard time to get the requirements installed together with the requirements of an other package (in this case MPES).
> 
> The requirements.txt is forcing conflicting versions, and therefore fails. The poetry installation is still very counterintuitive to me, I cannot seem to find a way to make it work as intended.
> 
> I will continue trying, but this issue should be closed only once a well tested environment installation procedure has been identified and described.
> 
> Also, a conda install .yaml would be really helpful.

I tested on it's own but not with another package. I agree that setting up a working envrionment with other packages using poetry is difficult. I really like the use of poetry for development but for deployment, I do agree it's unintuitive. To me, the best way still is to publish the package on pypi, which I was suggested we do later on. But setting a conda install file also makes sense. Perhaps a meeting is well deserved for this topic?

**Comment by steinnymir** [comment]
I agree that poetry is good for development, but we also need a "quick and simple" solution to showcase our package soon, where non-expert users can easily install and manage extra packages.

I believe the simplest solution is conda:
```
conda env create -f env.yml
conda activate sed_conda
```
followed by installation instructions for jupyter notebooks:
```
conda install ipykernel
python -m ipykernel install --user --name=sed_conda
```
I just tested this and seems to work...
I will add a pull request with the env.yml I (manually) created

**Comment by steinnymir** [comment]
Addressing this issue in #58

**Comment by rettigl** [comment]
This still enforces specific version of all packages. Could we have two setups, e.g. one for development based on poetry, where package versions are fixed, and one for pip where package versions are defined by lower limits?

**Comment by rettigl** [comment]
closed by PR #82
